posts/2024-03-18_xportr_0_4_0/xportr_0_4_0.qmd

```diff
@@ -0,0 +1,112 @@
+
+# How does it work?
+
+You too can experience some of the benefits of [`xportr`](https://atorus-research.github.io/xportr/) by taking a [`deep dive`](https://atorus-research.github.io/xportr/articles/deepdive.html) at [`xportr`](https://atorus-research.github.io/xportr/)!
+
+There are data deliverables and supporting documentation needed for a successful submission to Health Authorities, while you can find a walk through on how you can use `xportr` [`here`](https://atorus-research.github.io/xportr/articles/deepdive.html#what-goes-in-a-submission-to-a-health-authority), let's look at a high level example:
+
+```{r,  message = FALSE, echo = FALSE }
+library(xportr)
+library(magrittr)
+```
+
+Looking at an `adsl` and the define metadata information
+```{r,  message = FALSE, warnings = FALSE }
+adsl <- data.frame(
+  Subj = as.character(123, 456, 789),
+  Different = c("a", "b", "c"),
+  Val = c("1", "2", "3"),
+  Param = c("param1", "param2", "param3")
+)
+```
+
+and the define metadata information
+```{r,  message = FALSE, warnings = FALSE}
+metadata <- data.frame(
+  dataset = "test",
+  variable = c("Subj", "Param", "Val", "NotUsed"),
+  type = c("numeric", "character", "numeric", "character"),
+  format = NA,
+  order = c(1, 3, 4, 2)
+)
+```
+
+Users can use the [`xportr_metadata()`](https://atorus-research.github.io/xportr/reference/xportr_metadata.html) function along with [`xportr_type()`](https://atorus-research.github.io/xportr/reference/xportr_type.html) and [`xportr_order()`](https://atorus-research.github.io/xportr/reference/xportr_order.html) to not only apply metadata attributes to your data, receive feedbacks on your data based on your spec, all while also performing valuable checks.
+
+```{r, warnings = FALSE}
+adsl %>%
+  xportr_metadata(metadata, "test") %>%
+  xportr_type() %>%
+  xportr_order()
+```
+
+You can also use additional functions like [`xportr_write()`](https://atorus-research.github.io/xportr/reference/xportr_write.html) to generate your `xpt` file.
+
+# Key Features of [`xportr`](https://atorus-research.github.io/xportr/):
+
+- **Efficient Handling**: [`xportr`](https://atorus-research.github.io/xportr/) offers efficient avenues to produce `xpt` files, allowing users to manipulate clinical datasets with ease and export in `xpt` format in R.
+- **Data Integrity**: [`xportr`](https://atorus-research.github.io/xportr/) preserves data integrity by retaining variable labels, formats, and other metadata during data export operations.
+- **User-Friendly Interface**: [`xportr`](https://atorus-research.github.io/xportr/) provides a user-friendly interface with intuitive functions for generating `xpt` datasets and performing data manipulation tasks.
+
+# About the other kid on the blog, (`JSON`)! Is The Future `XPT` or JSON?
```

Comment:
Haha! Amazing!  Can you make a reference to the datasetjson blog post on Atorus' site and the datasetjson package?  

@mstackhouse maybe a follow up blog post on pharamverse is needed about datasetjson??